include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -1973,13 +1975,20 @@ namespace internal
               for (unsigned int c = 0;
                    c < GeometryInfo<dim>::max_children_per_cell;
                    ++c)
-                for (unsigned int i = 0; i < n_dofs_per_cell; ++i)
-                  for (unsigned int j = 0; j < n_dofs_per_cell; ++j)
-                    transfer.schemes[1].prolongation_matrix
-                      [i * n_dofs_per_cell *
-                         GeometryInfo<dim>::max_children_per_cell +
-                       j + c * n_dofs_per_cell] =
-                      fe.get_prolongation_matrix(c)(j, i);
+                {
+                  const auto matrix =
+                    reference_cell == ReferenceCells::Tetrahedron ?
+                      fe.get_prolongation_matrix(c, RefinementCase<dim>(1)) :
```

Comment:
I guess you will extend this in a follow-up PR?

Reply:
Yes, this is just a fix that will not break the current implementation, and this returns the same results as of right now. To not extend the scope of this PR, it will follow in the future.